generators/gradle/index.cjs

```diff
@@ -0,0 +1,135 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/* eslint-disable consistent-return */
+const chalk = require('chalk');
+const { mixBlueprintGenerator } = require('generator-jhipster/support');
+
+const { GENERATOR_PROJECT_NAME, GENERATOR_JAVA, GENERATOR_GRADLE } = require('../generator-list');
+const { files } = require('./files.cjs');
+const { GRADLE, GRADLE_VERSION } = require('./constants.cjs');
+const { BUILD_TOOL } = require('../java/constants.cjs');
+
+const MixedGenerator = mixBlueprintGenerator(GENERATOR_PROJECT_NAME, GENERATOR_JAVA);
+
+module.exports = class extends MixedGenerator {
+  constructor(args, opts, features) {
+    super(args, opts, { jhipsterModular: true, unique: 'namespace', ...features });
+
+    // Register options available to cli.
+    if (!this.fromBlueprint) {
+      this.registerCommonOptions();
+      this.registerProjectNameOptions();
+    }
+
+    if (this.options.help) return;
+
+    this.config.defaults({
+      [BUILD_TOOL]: GRADLE,
+    });
+
+    if (this.options.defaults) {
+      this.configureProjectName();
+      this.configureJava();
+    }
+
+    // Fallback to server templates to avoid duplications.
+    // TODO v8 move sources from server templates to gradle templates.
+    this.jhipsterTemplatesFolders.push(this.fetchFromInstalledJHipster('server', 'templates'));
+  }
+
+  async _beforeQueue() {
+    if (!this.fromBlueprint) {
+      await this.dependsOnJHipster(GENERATOR_JAVA, [], { configure: true });
+      await this.composeWithBlueprints(GENERATOR_GRADLE);
+    }
+  }
+
+  _initializing() {
+    return {
+      validateFromCli() {
+        this.checkInvocationFromCLI();
+      },
+      sayHello() {
+        if (!this.showHello()) return;
+        this.log(chalk.white('⬢ Welcome to the JHipster Gradle ⬢'));
+      },
+      loadRuntimeOptions() {
+        this.loadRuntimeOptions();
+      },
+      loadConstants() {
+        this.loadJavaConstants();
+      },
+    };
+  }
+
+  get initializing() {
+    if (this.delegateToBlueprint) return;
+    return this._initializing();
+  }
+
+  _configuring() {
+    return {
+      configure() {
+        this.configureProjectName();
+        this.configureJava();
+      },
+    };
+  }
+
+  get configuring() {
+    if (this.delegateToBlueprint) return;
+    return this._configuring();
+  }
+
+  _loading() {
+    return {
+      loadConstants() {
+        this.GRADLE_VERSION = GRADLE_VERSION;
+        this.loadJavaConstants();
```

Comment:
Same here for the loading part.
Ideally, we won't need any reference to any method of a parent generator.
We may be able to disable some of their queue execution if needed.
I would love to see the same philosophy as the one we have in the generated code:
- Reasonable default if not specified
- Custom behaviour if specified

Reply:
I tried to generate 3 generators for java (java, java-config, java-application).
But it became too bloated.

IMO cross-dependency should be avoided, but there are some cases it's not possible without modularizing too much.